### PR TITLE
typecheck svelte files in tests

### PR DESCRIPTION
**Pull request by Rich-Harris**
Adds `svelte-check` to the `check` script in packages/kit, which will hopefully help us catch more type errors early (https://github.com/sveltejs/kit/pull/1316#issuecomment-831491962).

This doesn't actually fix any of the surfaced errors, but it's low-hanging fruit if anyone wants to work on that

### Before submitting the PR, please make sure you do the following
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts

### CODE REVIEW

**Comment by Rich-Harris** [comment]
Made two changes to types:

* Added an internal `NormalizedLoadOutput` that mirrors the public `LoadOutput`, and changed `LoadOutput` to allow `string | Error` for `loaded.error`. An alternative would be to only allow `Error`
* Referenced Vite/Svelte types inside the Kit types. This means that people only need to reference Kit types in their own apps

File: packages/kit/src/runtime/client/types.d.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export type NavigationResult = {
 
 export type BranchNode = {
 	module: CSRComponent;
-	loaded: LoadOutput;
+	loaded: NormalizedLoadOutput;
```
Comment (dummdidumm) [inline packages/kit/src/runtime/client/types.d.ts:26]: If `NormalizedLoadOutput` is exported from the public types I wouldn't consider it an internal type and therefore move it into a public type, possibly `page`.

Comment (Rich-Harris) [reply packages/kit/src/runtime/client/types.d.ts:26]: where is it being exported from the public types?

Comment (dummdidumm) [reply packages/kit/src/runtime/client/types.d.ts:26]: Ok don't mind me, I'm stupid. I misread the file name and thought this was one of the public d.ts files.

**Pull request merged**